web/src/utils/iframes.ts

```diff
@@ -0,0 +1,190 @@
+import dimensions from '../constants/dimensions'
+
+export const EXTERNAL_SOURCES_COOKIE_NAME = 'Opt-In-External-Sources'
+export const addDoNotTrackParameter = (iframe: HTMLIFrameElement): void => {
+  if (iframe.src.includes('vimeo')) {
+    const url = new URL(iframe.src)
+    url.searchParams.append('dnt', '1')
+    iframe.setAttribute('src', url.href)
+  }
+}
+
+export const preserveIFrameSourcesFromContent = (
+  index: number,
+  source: string,
+  setExistingIframes: (sources: IframeSources) => void,
+  contentIframeSources: IframeSources,
+): void => {
+  const updatedContentSources: IframeSources = contentIframeSources
+  updatedContentSources[index] = source
+  setExistingIframes(updatedContentSources)
+}
+
+export const hideIframe = (iframe: HTMLIFrameElement): void => {
+  iframe.setAttribute('src', IFRAME_BLANK_SOURCE)
+  iframe.setAttribute('style', 'display:none')
+}
+
+export const restoreIframe = (iframe: HTMLIFrameElement, source: string): void => {
+  iframe.setAttribute('src', source)
+  iframe.setAttribute('style', 'display:block')
+}
+
+const getContainer = (element: HTMLElement, className: string, id: string): HTMLDivElement | null => {
+  if (document.getElementById(id)) {
+    return null
+  }
+  const container = document.createElement('div')
+  container.id = id
+  container.classList.add(className)
+  element.appendChild(container)
+  return container
+}
+
+const getIframeContainer = (
+  id: string,
+  viewportSmall: boolean,
+  iframe: HTMLIFrameElement,
+  deviceWidth: number,
+): HTMLDivElement => {
+  const existingContainer = document.getElementById(id)
+  if (existingContainer) {
+    return existingContainer as HTMLDivElement
+  }
+  const iframeContainer = document.createElement('div')
+  iframeContainer.classList.add('iframe-container')
+  iframeContainer.id = id
+  iframe.parentNode?.appendChild(iframeContainer)
+  iframeContainer.appendChild(iframe)
+  if (viewportSmall) {
+    // Scale the height depending on device width minus padding
+    const scaledHeight =
+      (deviceWidth / Number(iframe.width)) * Number(iframe.height) - dimensions.mainContainerHorizontalPadding
+    iframe.setAttribute('height', `${scaledHeight}`)
+  } else {
+    // Set the container width according to the iframe width
+    iframeContainer.setAttribute('style', `width:${iframe.width}px!important`)
+  }
+  return iframeContainer
+}
+
+const removeOptInContainer = (elementId: string) => {
+  const element = document.getElementById(elementId)
+  if (element) {
+    element.remove()
+  }
+}
+
+const showSource = (element: HTMLElement, source: string): void => {
+  const span = document.createElement('span')
+  span.classList.add('iframe-source')
+  element.appendChild(span)
+  span.appendChild(document.createTextNode(source))
+  element.appendChild(document.createElement('br'))
+}
+
+export const showMessage = (text: string, element: HTMLDivElement, iframeSource?: string): void => {
```

Comment:
`iframeSource` is never undefined, so you can just make it required and remove the check below.